dbms/src/Interpreters/JoinUtils.h

```diff
@@ -111,6 +126,26 @@ void computeDispatchHash(size_t rows,
                          std::vector<String> & partition_key_containers,
                          size_t join_restore_round,
                          WeakHash32 & hash);
+
+template <int>
+struct PointerTypeColumnHelper;
+
+template <>
+struct PointerTypeColumnHelper<4>
+{
+    using DataType = DataTypeInt32;
+    using ColumnType = ColumnVector<Int32>;
+    using ArrayType = PaddedPODArray<Int32>;
+};
+
+template <>
+struct PointerTypeColumnHelper<8>
```

Comment:
Will both `PointerTypeColumnHelper<4>` and `PointerTypeColumnHelper<8>` be used?

Reply:
No, 4 for 32-bit platform, 8 for 64-bit platform.